source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -91,7 +91,13 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
         return mut_status;
       }
     }
-    if (common_response.clear_route_cache()) {
+
+    bool allow_clearing_route_cache =
+        PROTOBUF_GET_WRAPPED_OR_DEFAULT(filter_.config(), allow_clearing_route_cache, true);
+    // Only clear the route cache if there is a mutation to the filter.
+    if (common_response.clear_route_cache() && allow_clearing_route_cache &&
+        (common_response.has_header_mutation() || common_response.has_body_mutation())) {
```

Comment:
The change of request header could affect the route selection so we need to clear route cache to trigger the route re-selection (so that we don't read from cache directly). The body mutation will not affect.
Can you remove has_body_mutation check?

Also, one nit. Can you move `allow_clearing_route_cache` to the first check. Slightly better for performance since we will not check the other conditions if first one failed.

Reply:
Removed body check and changed order.